server/intelligence-service/app/model.py

```diff
@@ -78,29 +78,13 @@ def call_model(state: State):
 app = workflow.compile(checkpointer=memory)
 
 
-def start_chat(input_message: str):
-    messages = [HumanMessage(input_message)]
-    state = State(messages=messages, step=0)
-    thread_id = "chat_" + str(randint(0, 9999))
-    config = {"configurable": {"thread_id": thread_id}}
-
-    output = app.invoke(
-        {"messages": state["messages"], "step": state["step"]},
-        config,
-    )
-
-    state["messages"] += output.get("messages", [])
-    return {"thread_id": thread_id, "state": state, "response": output}
-
-
-def chat(thread_id: str, input_message: str, state: State):
+def send_message(thread_id: str, input_message: str, state: State):
     config = {"configurable": {"thread_id": thread_id}}
     # append the new human message to the conversation
     state["messages"] += [HumanMessage(input_message)]
-    state["step"] += 1
 
     output = app.invoke(
```

Comment:
I think you would use a initial LLM call to choose between the step given the context and then use `conditional_edges` to select between the tasks in a second step. Each task node would then have a different prompt.